src/odemis/acq/move.py

```diff
@@ -390,11 +392,21 @@ def _doCryoSwitchSamplePosition(future, stage, align, target):
             if current_label == LOADING:
                 # As moving from loading position requires re-referencing the stage, move all axes together to
                 # prevent invalid/reachable position error
-                sub_moves.append((stage, filter_dict({'x', 'y', 'z'}, target_pos[target])))
+                sub_moves.append((stage, filter_dict({'y'}, target_pos[target])))
+                sub_moves.append((stage, filter_dict({'z'}, target_pos[target])))
+                sub_moves.append((stage, filter_dict({'rx', 'rz'}, target_pos[target])))
+                sub_moves.append((stage, filter_dict({'x'}, target_pos[target])))
```

Comment:
Here we need a comment too. I guess the idea is essentially the same/reverse as for going to LOADING: do the small movements first near the loading position, and end with the large x move to get close to the pole-piece.